api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -840,6 +840,12 @@ message HttpFilter {
     // Extension configs delivered through this mechanism are not expected to require warming (see https://github.com/envoyproxy/envoy/issues/12061).
     config.core.v3.ExtensionConfigSource config_discovery = 5;
   }
+
+  // If true, clients that do not support this filter may ignore the
+  // filter but otherwise accept the config.
+  // Otherwise, clients that do not support this filter must reject the config.
+  // [#not-implemented-hide:]
+  bool is_optional = 6;
```

Comment:
What does the rest of the implementation look like?  Is there a natural place to ignore the config for optional filters that are not built into the binary?

Reply:
@htuch said this probably wouldn't be hard to implement.

I'm guessing that we'll do it when validating the LDS response.  Presumably, that's the point at which we're checking that we have a filter registered for the type of the config in the `typed_config` field, and we'll NACK the config if we don't have one.  The purpose of the `is_optional` field is to have that check ignore the filter instead of NACKing the config.